functions-and-scope.js

```diff
@@ -0,0 +1,150 @@
+// Log het antwoord in de terminal.
+
+// ---- Verwachte uitkomsten:
+// cumLaude(grades) geeft 6
+// cumLaude([6, 4, 5]) geeft 0
+// cumLaude([8, 9, 4, 6, 10]) geeft 3
+
+
+/* Opdracht  2: Gemiddeld cijfer */
+
+/* 2a: Script schrijven  */
+// De studenten-administratie moet ieder blok opnieuw berekenen wat het gemiddelde eindcijfer is, maar we beginnen met de grades array van hierboven.
+// Schrijf de stapjes om dit te kunnen berekenen eerst uit en vraag jezelf de volgende dingen af:
+// * Hoe wordt een gemiddelde berekend?
+// * Wat moet ik verzamelen uit de array van cijfers om uiteindelijk een gemiddelde te kunnen berekenen?
+// * Hoe zorgt ik ervoor dat ik alle waardes uit de array kan langslopen, ook als de array wel 100 entries zou bevatten?
+// Log het antwoord in de terminal.
+
+let sum = 0;
+let gradesLength = grades.length;
+for (let i = 0; i < grades.length; i++) {
+    sum += grades[i];
+}
+console.log(sum / gradesLength);
+
+
+// ---- Verwachte uitkomst: 6.642857142857143
+
+
+/* 2b: Omschrijven tot een herbruikbare functie */
+// Schrijf een functie genaamd averageGrade, die een array van cijfers verwacht (zoals grades) en het gemiddelde cijfer teruggeeft. Gebruik hiervoor jouw antwoord van 2a.
+// Zorg ervoor dat jouw functie ook werkt als we een andere array willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
+// Log het antwoord in de terminal.
+
+
+function averageGrade(array) {
+    let sum = 0;
+    let arrayLength = array.length;
+    for (let i = 0; i < array.length; i++) {
+        sum += array[i];
+    }
+    return (sum / arrayLength).toFixed(2);
+}
+
+console.log(averageGrade(grades));
+console.log(averageGrade([6, 4, 5]));
+console.log(averageGrade([8, 9, 4, 6, 10]));
+
+// ---- Verwachte uitkomsten:
+// averageGrade(grades) geeft 6.642857142857143
+// averageGrade([6, 4, 5]) geeft xxxx
+// averageGrade([8, 9, 4, 6, 10]) geeft xxxx
+
+
+/* 2c: Afronden op twee decimalen */
+// Zorg ervoor dat het gemiddelde cijfer dat wordt teruggegeven uit de functie netjes wordt afgerond op twee decimalen.
+// Tip: Google is your best friend!
+
+
+/* Bonusopdracht: hoogste cijfer */
+
+/* 3a: Script schrijven  */
+// Schrijf een script die op basis van de grades array (hierboven) checkt wat het hoogst behaalde cijfer is. Je mag hier geen bestaande methoden voor gebruiken. Schrijf de stapjes eerst uit en vraag jezelf de volgende dingen af:
+// * Hoe kan ik iedere waarde van de array langsgaan?
+// * Op welke conditie moet ik checken?
+// * Hoe zorg ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan?
+// Log het antwoord in de terminal.
+
+let biggestNumber = 0;
+for (let i = 0; i < grades.length; i++) {
+    if (grades[i] > biggestNumber) {
+        biggestNumber = grades[i];
+    }
+}
+console.log(biggestNumber);
+
+// ---- Verwachte uitkomst: 9
+
+
+function highestGrade(array) {
+    let biggestNumber = 0;
```

Comment:
Het is beter om als default waarde altijd het eerste cijfer van de array te pakken. Cijfers zullen in dit voorbeeld altijd hoger dan 0 zijn, maar voer je functie eens uit met een array als `[-1, -2, -4, -1]`. Dan zal er `0` uitkomen terwijl die waarde niet eens in de array voorkomt 😉